plaso/engine/artifacts_trie.py

```diff
@@ -0,0 +1,299 @@
+    node = node.children[path_separator]
+    
+    # Handle the case when the input path is equal to the path_separator.
+    if path == path_separator:
+      node.artifacts_names.append(artifact_name)
+      return
+
+    path_segments = path.strip(path_separator).split(path_separator)
+    for segment in path_segments:
+      # Store the path_separator for each node.
+      if not hasattr(node, 'path_separator'):
+        node.path_separator = path_separator
+
+      if segment not in node.children:
+        node.children[segment] = TrieNode(path_separator)
+      node = node.children[segment]
+    node.artifacts_names.append(artifact_name)
+
+  def GetMatchingArtifacts(self, path, path_separator):
+    """Retrieves the artifact names that match the given path.
+
+    Args:
+        path (str): path to match against the Trie.
+        path_separator (str): path separator.
+
+    Returns:
+      list[str]: artifact names that match the path.
+    """
+    # Start at the root's child that matches the path_separator.
+    if path_separator not in self.root.children:
+      return []
+
+    sub_root_node = self.root.children[path_separator]
+    
+    # Handle the case when the input path is equal to the path_separator.
+    if path == path_separator:
+      matching_artifacts = set()
+      if sub_root_node.artifacts_names:
+        matching_artifacts.update(sub_root_node.artifacts_names)
+      return list(matching_artifacts)
+    path_segments = path.strip(path_separator).split(path_separator)
+    matching_artifacts = set()
+    # Update self.artifacts_paths before starting the search.
+    self.artifacts_paths = self._GetArtifactsPaths(sub_root_node)
+
+    def _search_trie(node, current_path, segments):
+      """Searches the trie for paths matching the given path segments.
+
+      Args:
+        node (TrieNode): current trie node being traversed.
+        current_path (str): path represented by the current node.
+        segments (list[str]): remaining path segments to match.
+      """
+      if node.artifacts_names:
+        for artifact_name in node.artifacts_names:
+          for artifact_path in self.artifacts_paths.get(artifact_name, []):
+            if self._ComparePathIfSanitized(
+                current_path, path_separator, artifact_path,
+                node.path_separator):
+              matching_artifacts.add(artifact_name)
+            elif glob.has_magic(artifact_path):
+              if self._MatchesGlobPattern(
+                  artifact_path, current_path, node.path_separator):
+                matching_artifacts.add(artifact_name)
+
+      if not segments:
+        return
+
+      segment = segments[0]
+      remaining_segments = segments[1:]
+
+      # Handle glob characters in the current segment.
+      for child_segment, child_node in node.children.items():
+        if (
+            child_segment == segment or
+            # comapring the sanitized version of the path segment stored in
```

Comment:
comapring => comparing.

could you restructure this code

(1) move nested function calls to separate lines
(2) combine the comments into 1 leading comment

Reply:
Done